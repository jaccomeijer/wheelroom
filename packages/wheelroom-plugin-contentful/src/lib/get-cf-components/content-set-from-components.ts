import {
  FieldType,
  WheelroomComponent,
  WheelroomComponents,
} from '@wheelroom/wheelroom'
import {
  ContentfulContent,
  ContentfulContentField,
  ContentfulContentSet,
} from '../../types/content-set'

export const cfContentSetFromComponents = (
  wrComponents: WheelroomComponents
): ContentfulContentSet => {
  const contentSet: ContentfulContentSet = []

  Object.entries(wrComponents).forEach(
    ([componentName, wrComponent]: [string, WheelroomComponent]) => {
      const content: ContentfulContent = {
        componentId: componentName,
        dependsOnModels: [],
        fields: {},
        model: componentName,
      }
      Object.entries(wrComponent.fields).forEach(
        ([fieldName, field]: [string, FieldType]) => {
          if ('allowedComponents' in field) {
            if (Array.isArray(field.allowedComponents)) {
              content.dependsOnModels.push(...field.allowedComponents)
            } else {
              content.dependsOnModels.push(field.allowedComponents)
            }
          }
          content.fields[fieldName] = {
            initialContent: field.initialContent,
            type: field.type,
          } as ContentfulContentField
        }
      )
      contentSet.push(content)
    }
  )
  return contentSet
}
