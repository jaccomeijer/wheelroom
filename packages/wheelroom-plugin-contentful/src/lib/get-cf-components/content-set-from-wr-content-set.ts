import { WheelroomComponents } from '@wheelroom/wheelroom'
import {
  ContentfulContent,
  ContentfulContentField,
  ContentfulContentSet,
  WheelroomContent,
  WheelroomContentSet,
} from '../../types/content-set'

export const cfContentSetFromWrContentSet = (
  wrContentSet: WheelroomContentSet,
  wrComponents: WheelroomComponents
): ContentfulContentSet => {
  const contentSet: ContentfulContentSet = []

  Object.entries(wrContentSet).forEach(
    ([componentId, wrContent]: [string, WheelroomContent]) => {
      const content: ContentfulContent = {
        componentId,
        dependsOnModels: [],
        fields: {},
        model: wrContent.model,
      }
      Object.entries(wrContent.fields).forEach(
        ([fieldName, initialContent]: [string, string]) => {
          if (!(wrContent.model in wrComponents)) {
            // Most probably the filter argument is active
            return
          }
          const wrField = wrComponents[wrContent.model].fields[fieldName]
          if (
            wrField &&
            typeof wrField === 'object' &&
            'allowedComponents' in wrField
          ) {
            if (Array.isArray(wrField.allowedComponents)) {
              content.dependsOnModels.push(...wrField.allowedComponents)
            } else {
              content.dependsOnModels.push(wrField.allowedComponents)
            }
          }
          content.fields[fieldName] = {
            initialContent,
            type: wrField.type,
          } as ContentfulContentField
        }
      )
      contentSet.push(content)
    }
  )
  return contentSet
}
