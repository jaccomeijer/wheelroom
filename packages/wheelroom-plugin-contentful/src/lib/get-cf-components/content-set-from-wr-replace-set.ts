import { WheelroomComponents } from '@wheelroom/wheelroom'
import {
  ContentfulContent,
  ContentfulContentField,
  ContentfulContentSet,
} from '../../types/content-set'
import {
  WheelroomReplaceSet,
  ReplaceDetails,
  ReplaceFields,
} from '../../types/replace-sets'

export const cfContentSetFromWrReplaceSet = (
  wrReplaceSet: WheelroomReplaceSet,
  wrComponents: WheelroomComponents
): ContentfulContentSet => {
  const contentSet: ContentfulContentSet = []

  Object.entries(wrReplaceSet).forEach(
    ([model, wrFields]: [string, ReplaceFields]) => {
      const content: ContentfulContent = {
        componentId: '',
        dependsOnModels: [],
        fields: {},
        model,
      }
      Object.entries(wrFields.fields).forEach(
        ([fieldName, replaceDetails]: [string, ReplaceDetails]) => {
          if (!(model in wrComponents)) {
            // Most probably the filter argument is active
            return
          }
          const wrField = wrComponents[model].fields[fieldName]
          content.fields[fieldName] = {
            initialContent: replaceDetails,
            type: wrField.type,
          } as ContentfulContentField
        }
      )
      contentSet.push(content)
    }
  )
  return contentSet
}
