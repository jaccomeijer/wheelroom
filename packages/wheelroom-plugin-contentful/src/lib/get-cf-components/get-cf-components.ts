import {
  FieldType,
  WheelroomComponent,
  WheelroomComponents,
} from '@jacco-meijer/wheelroom'
import {
  ContentfulContent,
  ContentfulContentSet,
} from '../../types/content-set'
import { ContentfulComponents } from '../../types/contentful-components'
import { ContentfulFieldDefinitions } from '../../types/contentful-field-definitions'
import { getCfComponent } from './get-cf-component'

export const getCfComponents = (
  wrComponents: WheelroomComponents,
  fieldDefinitions: ContentfulFieldDefinitions,
  cfContentSet: ContentfulContentSet
): ContentfulComponents => {
  const cfComponents: ContentfulComponents = []

  // Loop through all content, set each field initialContent and process with
  // matching component
  cfContentSet.forEach((cfContent: ContentfulContent) => {
    if (!(cfContent.model in wrComponents)) {
      // Most probably the filter argument is active
      return
    }
    const matchingWrComponent = wrComponents[cfContent.model]
    const wrComponentWithContent: WheelroomComponent = {
      fields: {},
      modelVersion: matchingWrComponent.modelVersion,
      settings: matchingWrComponent.settings,
    }
    Object.entries(matchingWrComponent.fields).forEach(
      ([fieldName, field]: [string, FieldType]) => {
        // Copy field into new object and set initialContent
        wrComponentWithContent.fields[fieldName] = Object.assign({}, field, {
          initialContent: cfContent.fields[fieldName]
            ? cfContent.fields[fieldName].initialContent
            : undefined,
        })
      }
    )
    const cfComponent = getCfComponent({
      componentId: cfContent.componentId,
      componentName: cfContent.model,
      fieldDefinitions,
      wrComponent: wrComponentWithContent,
      wrComponents,
    })
    cfComponents.push(cfComponent)
  })
  return cfComponents
}
