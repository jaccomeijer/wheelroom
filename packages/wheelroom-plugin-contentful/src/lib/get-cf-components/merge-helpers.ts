// Helper to add object or array if needed
export const createIfMissing = (
  base: any,
  key: string,
  what: 'object' | 'array'
) => {
  if (key in base) {
    return base
  } else {
    base[key] = what === 'object' ? {} : []
    return base
  }
}
