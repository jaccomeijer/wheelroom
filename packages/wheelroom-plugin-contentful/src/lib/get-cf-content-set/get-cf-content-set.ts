import { WheelroomComponents } from '@wheelroom/wheelroom'
import {
  WheelroomContentSet,
  ContentfulContentSet,
} from '../../types/content-set'
import { cfContentSetFromComponents } from '../get-cf-components/content-set-from-components'
import { cfContentSetFromWrContentSet } from '../get-cf-components/content-set-from-content-set'

export const getCfContentSet = (
  wrComponents: WheelroomComponents,
  wrContentSet?: WheelroomContentSet
): ContentfulContentSet => {
  // If no content is set, create one based on component names and initial
  // content
  let cfContentSet: ContentfulContentSet
  if (wrContentSet) {
    cfContentSet = cfContentSetFromWrContentSet(wrContentSet, wrComponents)
  } else {
    cfContentSet = cfContentSetFromComponents(wrComponents)
  }
  return cfContentSet
}
