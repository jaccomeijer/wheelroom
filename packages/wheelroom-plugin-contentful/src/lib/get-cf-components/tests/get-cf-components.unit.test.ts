import { getFilteredComponents } from '@wheelroom/wheelroom'
import { argvCommand } from '../../../fixtures/argv-command'
import { argvContentSets } from '../../../fixtures/argv-content-sets'
import { getCfComponents } from '../get-cf-components'
import { getCfContentSet } from '../../get-cf-content-set/get-cf-content-set'
import { orderByDependency } from '../../get-cf-content-set/order-by-dependency'

describe('Generate contentful components should', () => {
  test('match the sample set without content', () => {
    const pluginOptions =
      argvCommand.options['@wheelroom/wheelroom-plugin-contentful']
    const wheelroomComponents = getFilteredComponents(argvCommand)

    const cfContentSet = getCfContentSet(wheelroomComponents)
    orderByDependency(cfContentSet)
    const contentfulComponents = getCfComponents(
      wheelroomComponents,
      pluginOptions.fieldDefinitions,
      cfContentSet
    )
    expect(contentfulComponents).toMatchSnapshot()
  })

  test('match the sample set with a content set', () => {
    const pluginOptions =
      argvCommand.options['@wheelroom/wheelroom-plugin-contentful']
    const wheelroomComponents = getFilteredComponents(argvCommand)

    const cfContentSet = getCfContentSet(
      wheelroomComponents,
      argvContentSets.test
    )
    orderByDependency(cfContentSet)
    const contentfulComponents = getCfComponents(
      wheelroomComponents,
      pluginOptions.fieldDefinitions,
      cfContentSet
    )
    expect(contentfulComponents).toMatchSnapshot()
  })
})
