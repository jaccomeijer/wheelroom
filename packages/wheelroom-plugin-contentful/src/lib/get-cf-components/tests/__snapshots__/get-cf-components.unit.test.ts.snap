// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generate contentful components should match the sample set with a content set 1`] = `
Array [
  Object {
    "componentId": "someSubPage",
    "description": "Sub page model",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "Testing some sub page",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "subPageModel",
  },
  Object {
    "componentId": "secondSection",
    "description": "Second section",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "testing second section model",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "secondSection",
  },
  Object {
    "componentId": "noFragment",
    "description": "No fragment",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "Testing no fragment model",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "noFragment",
  },
  Object {
    "componentId": "globals",
    "description": "Global model",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "Testing global model",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "globalModel",
  },
  Object {
    "componentId": "firstSectionC",
    "description": "First section",
    "displayField": "title",
    "fields": Object {
      "defaultField": Object {
        "createContentData": "Third content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Default field",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "C - Testing first section model",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "firstSection",
  },
  Object {
    "componentId": "firstSectionB",
    "description": "First section",
    "displayField": "title",
    "fields": Object {
      "defaultField": Object {
        "createContentData": "Second content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Default field",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "B - Testing first section model",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "firstSection",
  },
  Object {
    "componentId": "firstSectionA",
    "description": "First section",
    "displayField": "title",
    "fields": Object {
      "defaultField": Object {
        "createContentData": "First content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Default field",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "A - Testing first section model",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "firstSection",
  },
  Object {
    "componentId": "anotherPage",
    "description": "Page model",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "Testing another page model",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "pageModel",
  },
  Object {
    "componentId": "allFieldTypesPage2",
    "description": "All field types",
    "displayField": "title",
    "fields": Object {
      "dateField": Object {
        "createContentData": "2020-06-03T00:00+01:00",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Date field",
          "required": false,
          "type": "Date",
        },
        "widgetId": "datePicker",
      },
      "dropdownField": Object {
        "createContentData": "item B",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Dropdown field",
          "required": false,
          "type": "Symbol",
          "validations": Array [
            Object {
              "in": Array [
                "item A",
                "item B",
                "item C",
              ],
            },
          ],
        },
        "widgetId": "dropdown",
      },
      "imageField": Object {
        "createContentData": "demoAsset",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "linkType": "Asset",
          "localized": false,
          "name": "Image field",
          "required": false,
          "type": "Link",
          "validations": Array [
            Object {
              "linkMimetypeGroup": Array [
                "image",
              ],
            },
          ],
        },
        "widgetId": "assetLinkEditor",
      },
      "longTextField": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Long text field",
          "required": false,
          "type": "Text",
        },
        "widgetId": "multipleLine",
      },
      "multipleComponentsField": Object {
        "createContentData": Array [
          "firstSectionB",
          "secondSection",
        ],
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "items": Object {
            "linkType": "Entry",
            "type": "Link",
            "validations": Array [
              Object {
                "linkContentType": Array [
                  "page",
                  "firstSection",
                  "secondSection",
                ],
              },
            ],
          },
          "localized": false,
          "name": "Multiple components field",
          "required": false,
          "type": "Array",
        },
        "widgetId": "entryLinksEditor",
      },
      "numberField": Object {
        "createContentData": 25,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Number field",
          "required": false,
          "type": "Integer",
        },
        "widgetId": "numberEditor",
      },
      "richTextField": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Rich text field",
          "required": false,
          "type": "RichText",
        },
        "widgetId": "richTextEditor",
      },
      "shortTextField": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Short text field",
          "required": false,
          "type": "Symbol",
          "validations": Array [
            Object {
              "size": Object {
                "max": 155,
                "min": 0,
              },
            },
          ],
        },
        "widgetId": "singleLine",
      },
      "shortTextFieldExtended": Object {
        "createContentData": "onlyCamelCaseHere",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Short text field extended",
          "required": false,
          "type": "Symbol",
          "validations": Array [
            Object {
              "message": "Please use camel case: onlyCamelCaseAllowed",
              "regexp": Object {
                "pattern": "^[a-z]+([A-Z][a-z0-9]+)*$",
              },
            },
            Object {
              "size": Object {
                "max": 155,
                "min": 0,
              },
            },
          ],
        },
        "widgetId": "singleLine",
      },
      "singleComponentField": Object {
        "createContentData": "anotherPage",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "linkType": "Entry",
          "localized": false,
          "name": "Single component field",
          "required": false,
          "type": "Link",
          "validations": Array [
            Object {
              "linkContentType": Array [
                "pageModel",
              ],
            },
          ],
        },
        "widgetId": "entryLinkEditor",
      },
      "tagsField": Object {
        "createContentData": Array [
          "some",
          "tags",
        ],
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "items": Object {
            "type": "Symbol",
          },
          "localized": false,
          "name": "Tags field",
          "required": false,
          "type": "Array",
        },
        "widgetId": "tagEditor",
      },
      "title": Object {
        "createContentData": "Testing allFieldTypesPage #2",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "allFieldTypes",
  },
  Object {
    "componentId": "allFieldTypesPage1",
    "description": "All field types",
    "displayField": "title",
    "fields": Object {
      "dateField": Object {
        "createContentData": "2019-06-03T00:00+01:00",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Date field",
          "required": false,
          "type": "Date",
        },
        "widgetId": "datePicker",
      },
      "dropdownField": Object {
        "createContentData": "item A",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Dropdown field",
          "required": false,
          "type": "Symbol",
          "validations": Array [
            Object {
              "in": Array [
                "item A",
                "item B",
                "item C",
              ],
            },
          ],
        },
        "widgetId": "dropdown",
      },
      "imageField": Object {
        "createContentData": "demoAsset",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "linkType": "Asset",
          "localized": false,
          "name": "Image field",
          "required": false,
          "type": "Link",
          "validations": Array [
            Object {
              "linkMimetypeGroup": Array [
                "image",
              ],
            },
          ],
        },
        "widgetId": "assetLinkEditor",
      },
      "longTextField": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Long text field",
          "required": false,
          "type": "Text",
        },
        "widgetId": "multipleLine",
      },
      "multipleComponentsField": Object {
        "createContentData": Array [
          "firstSectionA",
          "secondSection",
        ],
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "items": Object {
            "linkType": "Entry",
            "type": "Link",
            "validations": Array [
              Object {
                "linkContentType": Array [
                  "page",
                  "firstSection",
                  "secondSection",
                ],
              },
            ],
          },
          "localized": false,
          "name": "Multiple components field",
          "required": false,
          "type": "Array",
        },
        "widgetId": "entryLinksEditor",
      },
      "numberField": Object {
        "createContentData": 25,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Number field",
          "required": false,
          "type": "Integer",
        },
        "widgetId": "numberEditor",
      },
      "richTextField": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Rich text field",
          "required": false,
          "type": "RichText",
        },
        "widgetId": "richTextEditor",
      },
      "shortTextField": Object {
        "createContentData": "Content for testing this field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Short text field",
          "required": false,
          "type": "Symbol",
          "validations": Array [
            Object {
              "size": Object {
                "max": 155,
                "min": 0,
              },
            },
          ],
        },
        "widgetId": "singleLine",
      },
      "shortTextFieldExtended": Object {
        "createContentData": "onlyCamelCaseHere",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Short text field extended",
          "required": false,
          "type": "Symbol",
          "validations": Array [
            Object {
              "message": "Please use camel case: onlyCamelCaseAllowed",
              "regexp": Object {
                "pattern": "^[a-z]+([A-Z][a-z0-9]+)*$",
              },
            },
            Object {
              "size": Object {
                "max": 155,
                "min": 0,
              },
            },
          ],
        },
        "widgetId": "singleLine",
      },
      "singleComponentField": Object {
        "createContentData": "anotherPage",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "linkType": "Entry",
          "localized": false,
          "name": "Single component field",
          "required": false,
          "type": "Link",
          "validations": Array [
            Object {
              "linkContentType": Array [
                "pageModel",
              ],
            },
          ],
        },
        "widgetId": "entryLinkEditor",
      },
      "tagsField": Object {
        "createContentData": Array [
          "some",
          "tags",
        ],
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "items": Object {
            "type": "Symbol",
          },
          "localized": false,
          "name": "Tags field",
          "required": false,
          "type": "Array",
        },
        "widgetId": "tagEditor",
      },
      "title": Object {
        "createContentData": "Testing allFieldTypesPage #1",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "allFieldTypes",
  },
]
`;

exports[`Generate contentful components should match the sample set without content 1`] = `
Array [
  Object {
    "componentId": "subPageModel",
    "description": "Sub page model",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": undefined,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "Sub page model generated from boilerplate",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "subPageModel",
  },
  Object {
    "componentId": "secondSection",
    "description": "Second section",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": undefined,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "Second section generated from boilerplate",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "secondSection",
  },
  Object {
    "componentId": "pageModel",
    "description": "Page model",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": undefined,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "Page model generated from boilerplate",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "pageModel",
  },
  Object {
    "componentId": "noFragment",
    "description": "No fragment",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": undefined,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "No fragment generated from boilerplate",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "noFragment",
  },
  Object {
    "componentId": "globalModel",
    "description": "Global model",
    "displayField": "title",
    "fields": Object {
      "heading": Object {
        "createContentData": undefined,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Heading",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "Global model generated from boilerplate",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "globalModel",
  },
  Object {
    "componentId": "firstSection",
    "description": "First section",
    "displayField": "title",
    "fields": Object {
      "defaultField": Object {
        "createContentData": undefined,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Default field",
          "required": false,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
      "title": Object {
        "createContentData": "First section generated from boilerplate",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "firstSection",
  },
  Object {
    "componentId": "allFieldTypes",
    "description": "All field types",
    "displayField": "title",
    "fields": Object {
      "dateField": Object {
        "createContentData": undefined,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Date field",
          "required": false,
          "type": "Date",
        },
        "widgetId": "datePicker",
      },
      "dropdownField": Object {
        "createContentData": "item A",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Dropdown field",
          "required": false,
          "type": "Symbol",
          "validations": Array [
            Object {
              "in": Array [
                "item A",
                "item B",
                "item C",
              ],
            },
          ],
        },
        "widgetId": "dropdown",
      },
      "imageField": Object {
        "createContentData": "demoAsset",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "linkType": "Asset",
          "localized": false,
          "name": "Image field",
          "required": false,
          "type": "Link",
          "validations": Array [
            Object {
              "linkMimetypeGroup": Array [
                "image",
              ],
            },
          ],
        },
        "widgetId": "assetLinkEditor",
      },
      "longTextField": Object {
        "createContentData": "This is a long text field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Long text field",
          "required": false,
          "type": "Text",
        },
        "widgetId": "multipleLine",
      },
      "multipleComponentsField": Object {
        "createContentData": Array [
          "firstSection",
          "secondSection",
        ],
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "items": Object {
            "linkType": "Entry",
            "type": "Link",
            "validations": Array [
              Object {
                "linkContentType": Array [
                  "page",
                  "firstSection",
                  "secondSection",
                ],
              },
            ],
          },
          "localized": false,
          "name": "Multiple components field",
          "required": false,
          "type": "Array",
        },
        "widgetId": "entryLinksEditor",
      },
      "numberField": Object {
        "createContentData": 5,
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Number field",
          "required": false,
          "type": "Integer",
        },
        "widgetId": "numberEditor",
      },
      "richTextField": Object {
        "createContentData": "This is a rich text field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Rich text field",
          "required": false,
          "type": "RichText",
        },
        "widgetId": "richTextEditor",
      },
      "shortTextField": Object {
        "createContentData": "This is a short text field",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Short text field",
          "required": false,
          "type": "Symbol",
          "validations": Array [
            Object {
              "size": Object {
                "max": 155,
                "min": 0,
              },
            },
          ],
        },
        "widgetId": "singleLine",
      },
      "shortTextFieldExtended": Object {
        "createContentData": "onlyCamelCaseAllowdForThisExtendedField",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "localized": false,
          "name": "Short text field extended",
          "required": false,
          "type": "Symbol",
          "validations": Array [
            Object {
              "message": "Please use camel case: onlyCamelCaseAllowed",
              "regexp": Object {
                "pattern": "^[a-z]+([A-Z][a-z0-9]+)*$",
              },
            },
            Object {
              "size": Object {
                "max": 155,
                "min": 0,
              },
            },
          ],
        },
        "widgetId": "singleLine",
      },
      "singleComponentField": Object {
        "createContentData": "pageModel",
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "linkType": "Entry",
          "localized": false,
          "name": "Single component field",
          "required": false,
          "type": "Link",
          "validations": Array [
            Object {
              "linkContentType": Array [
                "pageModel",
              ],
            },
          ],
        },
        "widgetId": "entryLinkEditor",
      },
      "tagsField": Object {
        "createContentData": Array [
          "tag A",
          "tag B",
        ],
        "settings": Object {
          "helpText": "",
        },
        "specs": Object {
          "items": Object {
            "type": "Symbol",
          },
          "localized": false,
          "name": "Tags field",
          "required": false,
          "type": "Array",
        },
        "widgetId": "tagEditor",
      },
      "title": Object {
        "createContentData": "All field types generated from boilerplate",
        "settings": Object {
          "helpText": "Never displayed, only used for listing within Contentful",
        },
        "specs": Object {
          "localized": false,
          "name": "Title",
          "required": true,
          "type": "Symbol",
        },
        "widgetId": "singleLine",
      },
    },
    "modelVersion": "1.0.0",
    "type": "allFieldTypes",
  },
]
`;
